reporter/v2/cmd/reporter/reconciler/reconcile.go

```diff
@@ -81,23 +82,33 @@ var ReconcileCmd = &cobra.Command{
 		}
 		cfg.SetDefaults()
 
+		broadcaster, stopBroadcast, err := reporter.NewEventBroadcaster(ctx, cfg)
+		if err != nil {
+			log.Error(err, "couldn't initialize event broadcaster")
+			os.Exit(1)
+		}
+
 		task, err := reporter.NewReconcileTask(
 			ctx,
 			cfg,
+			broadcaster,
 			reporter.Namespace(namespace),
 		)
 
 		if err != nil {
 			log.Error(err, "couldn't initialize task")
+			stopBroadcast()
 			os.Exit(1)
```

Comment:
We can make this a little clearer and easier for you to handle. Let's move stopBroadcast to the defer. We can change the command function to return an error `RunE`, then just return the errors here instead of os.Exit(1). Returning an error will cause it to exit so we don't need to call os.Exit any longer. Then we can remove the stopBroadcast() from the if's and end, it'll run when the function exits.